### DISCUSSION TITLE
ineffective x-init

### DISCUSSION THREAD

**Discussion by markfirmware** [alpinejs/alpine#936] [1. Help] [answered]
https://codepen.io/markfirmware/pen/wvzeGjX

When the fourth pin digit is entered, the x-init should change state to state1 but it remains state0.

**Comment by SimoTod** [comment, marked as answer]
It's syntactically invalid. X-init only works next to x-data. Note that, if you put x-data on that div, you won't be able to acess state in that way.

**Reply by markfirmware** [reply]
I see, thank you for the prompt response!

**Reply by SimoTod** [reply]
No worries. You can use a computed property to achieve something like that:
```html
<div x-data="{ pin: '', get state() { return this.pin.length < 4 ? 'state0' : 'state1' } }">
  <template x-if="pin.length != 4">
    <input x-model=pin placeholder="4 digit pin">
  </template>
    <span x-text=pin>
    </span>
    <span x-text=state>
    </span>
</div>
```
Or, alternative, you can use `$watch`
